pandas/core/indexes/range.py

```diff
@@ -30,23 +30,30 @@ class RangeIndex(Int64Index):
     """
     Immutable Index implementing a monotonic integer range.
 
-    RangeIndex is a memory-saving special case of Int64Index limited to
-    representing monotonic ranges. Using RangeIndex may in some instances
-    improve computing speed.
+    RangeIndex is a memory-saving special case of :class:`~pandas.Int64Index`
+    limited to representing monotonic ranges. Using RangeIndex may in some
+    instances improve computing speed.
 
     This is the default index type used
-    by DataFrame and Series when no explicit index is provided by the user.
+    by `DataFrame` and `Series` when no explicit index is provided by the user.
 
     Parameters
     ----------
-    start : int (default: 0), or other RangeIndex instance.
-        If int and "stop" is not given, interpreted as "stop" instead.
-    stop : int (default: 0)
-    step : int (default: 1)
-    name : object, optional
-        Name to be stored in the index
+    start : int or other RangeIndex instance, default 0
+        If of type `int` and `stop` is not given, interpreted as `stop`
+        instead.
+    stop : int, default 0
+    step : int, default 1
```

Comment:
It'f be good to have these documented. Also, `start`, which would be good to have a description, besides the behavior described.

Reply:
Re documenting `start`, `stop` and `step`: I also thought that it would be nice to document them, and figured we could just recycle the [python documentation for `range`](https://docs.python.org/3/library/stdtypes.html#typesseq-range). However, the documentation for the arguments is very basic in python as well: "`stop`: The value of the stop parameter". Maybe there is not much else to say?

But I'll try to adapt the rest of the python `range` docs (i.e. the long description) and try to come up with some useful examples.

Reply:
I think it can be useful to say whether the `stop` value is included or not. It's not obvious to me that `range(0, 10)` doesn't include the `10`.

Also, for beginner users, something as simple as  `start : The first value of the generated range.`  can be useful. `step` can be tricky, do we support negative values, as `range`?

I do think there is a lot to say. :)

Reply:
> think it can be useful to say whether the stop value is included or not. It's not obvious to me that range(0, 10) doesn't include the 10.

this is standard python behavior to NOT include the rhs on ranges.